src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java

```diff
@@ -543,8 +546,36 @@ public String apply(RuleExecutionContext context) {
         .findFirst()
         .orElse(STUB_FIELD_TYPE_ID);
     }
+  },
+
+  SET_AUTHORITY_SOURCE_FILE_ID() {
+    private static final String CODE_PARAMETER = "code";
+
+    @Override
+    public String apply(RuleExecutionContext context) {
+      var value = context.getRuleParameter().getString(CODE_PARAMETER);
+      var prefix = getPrefix(value);
+      var authoritySourceFiles = context.getMappingParameters().getAuthoritySourceFiles();
+
+      if (authoritySourceFiles == null || prefix == null) {
+        return null;
+      }
+
+      return authoritySourceFiles.stream()
+        .filter(authoritySourceFile -> authoritySourceFile.getCodes().stream()
+          .anyMatch(code -> code.equalsIgnoreCase(prefix)))
+        .map(AuthoritySourceFile::getId)
+        .findFirst()
+        .orElse(null);
+    }
```

Comment:
I think that the best solution for this is to find the longest substring-prefix. A possible solution is to have a map sorted by a key length that starts with the longest with key = code and value = id and then iterate this map until we found a prefix that fits a condition value.startsWith(code)